activation/nipost.go

```diff
@@ -91,19 +105,28 @@ type poetDbAPI interface {
 func NewNIPostBuilder(
 	nodeID types.NodeID,
 	postSetupProvider postSetupProvider,
-	poetProvers []PoetProvingServiceClient,
 	poetDB poetDbAPI,
+	poetServers []string,
 	dataDir string,
 	lg log.Log,
 	signer *signing.EdSigner,
 	poetCfg PoetConfig,
 	layerClock layerClock,
 	opts ...NIPostBuilderOption,
-) *NIPostBuilder {
+) (*NIPostBuilder, error) {
+	poetClients := make(map[string]PoetProvingServiceClient, len(poetServers))
+	for _, address := range poetServers {
+		client, err := NewHTTPPoetClient(address, poetCfg)
+		if err != nil {
+			return nil, fmt.Errorf("cannot create poet client: %w", err)
```

Comment:
curious. what's the benefit of parsing here vs parsing in node.go?

Reply:
Imo the node doesn't need to know about the dependencies of the `NIPoSTBuilder`. It passes in the configuration to the service and the service does with it what is needed. This way `newHTTPPoetClient` could even be private and the interface for it as well, since it is not needed outside the `activation` package.

Reply:
mixing critical change with non-critical change simply slow down the review process.

Reply:
I had to touch the code in question anyway. The field `poetProvers` was updated from `[]PoetProvers` to `map[string]PoetProvers` to be able to get one of the entries by Address (which wasn't possible before).

So either I add this code to iterate over `[]PoetProvers` call `Address()` on them and add them to the map, or I initialize them here directly. Seems like a related change to me 🤷.